jsettlers.graphics/src/main/java/jsettlers/graphics/map/controls/original/panel/content/settlers/profession/ProfessionPanel.java

```diff
@@ -1,66 +1,245 @@
+		this.uiContentUpdater.addListener(this);
+		this.lastChangeTimestamp = 0;
 	}
 
 	@Override
 	public UIPanel getPanel() {
 		return panel;
 	}
 
-	private IPartitionSettings currentDistributionSettingsProvider(IGraphicsGrid grid, ShortPoint2D position) {
-		if (grid.getPlayerAt(position.x, position.y) != null) {
-			return grid.getPartitionData(position.x, position.y).getPartitionSettings();
-		}
+	@Override
+	public ESecondaryTabType getTabs() {
+		return ESecondaryTabType.SETTLERS;
+	}
 
-		return null;
+	@Override
+	public void showMapPosition(ShortPoint2D position, IGraphicsGrid grid) {
+		this.panel.setPosition(position);
+		super.showMapPosition(position, grid);
+		this.uiContentUpdater.updatePosition(grid, position);
 	}
 
 	@Override
-	public ESecondaryTabType getTabs() {
-		return ESecondaryTabType.SETTLERS;
+	public void contentShowing(ActionFireable actionFireable) {
+		super.contentShowing(actionFireable);
+		this.uiContentUpdater.start();
 	}
 
 	@Override
-	public void update(IPartitionSettings data) {
+	public void update(IPartitionSettings partitionSettings) {
+		// Use delay to prevent race condition between direct ui update and settings update
+		if (System.currentTimeMillis() - lastChangeTimestamp > 5000) {
+			IProfessionSettings professionSettings = partitionSettings.getProfessionSettings();
+			this.panel.setup(professionSettings);
+		}
+	}
+	
+	public void setPlayer(IInGamePlayer player) {
+		uiContentUpdater.setPlayer(player);
+	}
 
+	private static IPartitionSettings currentDistributionSettingsProvider(IGraphicsGrid grid, ShortPoint2D position) {
+		return grid.getPlayerAt(position.x, position.y).getPlayerId() >= 0 ? grid.getPartitionData(position.x, position.y).getPartitionSettings() : null;
 	}
 
-	public enum ConfigurableRatioType {
-		BEARER,
-		DIGGER,
-		BRICKLAYER
+	public enum EProfessionType {
+		CARRIER("Carriers", true, EMovableType.BEARER),
+		DIGGER("Diggers", false, EMovableType.DIGGER),
+		BUILDER("Builders", false, EMovableType.BRICKLAYER);
+
+		public final String label;
+		public final boolean min;
+		public final EMovableType moveableType;
+
+		private EProfessionType(String label, boolean min, EMovableType movableType) {
+			this.label = label;
+			this.min = min;
+			this.moveableType = movableType;
+		}
+	}
+
+	public class ContentPanel extends Panel {
+		private ShortPoint2D position;
+		private final SettlerPanel diggerPanel;
+		private final SettlerPanel carrierPanel;
+		private final SettlerPanel builderPanel;
+
+		public ContentPanel() {
+			super(118f, 216f);
+			this.position = new ShortPoint2D(0, 0);
+			this.carrierPanel = new SettlerPanel(widthInPx, EProfessionType.CARRIER);
+			this.diggerPanel = new SettlerPanel(widthInPx, EProfessionType.DIGGER);
+			this.builderPanel = new SettlerPanel(widthInPx, EProfessionType.BUILDER);
+
+			add(Panel.box(new Label(Labels.getString("Settlers"), EFontSize.NORMAL), widthInPx, 20f), 0f, 0f);
```

Comment:
The Labels.getString is meant as a translation mechanism.
But there is not key "Settlers" defined (atleast not in labels_en.properties or labels_de.properties)
btw. the core game and the mapcreator use different labels_ files for some reason.

This label should probably be translated maybe place it below settler_stats_worker.
There is no general name convention for the labels_ files but I would prefer words as lowercase, _ instead of - and enums as uppercase.

Reply:
I have added the label to the properties.